### PR TITLE
Lai cai cover

### PR DESCRIPTION
**Pull request by rcnijzink**

### CODE REVIEW

File: VOM_Fortran/VOM-code/modules.f90 [hunk 1]
```diff
@@ -185,7 +185,7 @@ module vegmod
       REAL*8  :: wsnew                  ! Total soil water store at next time step
       REAL*8  :: wsold                  ! Previous total soil water storage
 
-      REAL*8  :: o_pct = 0.300000d+00   ! Projected cover perennial vegetation (0-1)
+      REAL*8  :: o_cai = 0.300000d+00   ! Projected cover perennial vegetation (0-1)
```
Comment (schymans) [inline VOM_Fortran/VOM-code/modules.f90]: Why is a number provided here? Should this not be in an input file and/or optimised? Same in L194-197.

File: VOM_Fortran/VOM-code/transpmodel.f90 [hunk 2]
```diff
@@ -978,11 +978,14 @@ subroutine vom_init_vegpar ()
...
 
-      q_md   = o_pct * i_mdtf + o_mdstore
+      q_md   = o_cai * i_mdtf + o_mdstore
       q_mqx  = q_md * i_mqxtf
       mqtnew = 0.95d0 * q_mqx                  ! initial wood water storage
       mqtold = mqtnew
       rsurftnew(:) = 0.d0
+      lai_lt(:) = 2.5d0 * (/1.0d0-i_incrlait,1.0d0,1.0d0+i_incrlait/)
+      lai_lg(:) = 2.5d0 * (/1.0d0-i_incrlaig,1.0d0,1.0d0+i_incrlaig/)
+
```
Comment (schymans) [inline VOM_Fortran/VOM-code/transpmodel.f90:990]: Why is the 2.5 still here? What does it mean?

File: VOM_Fortran/VOM-code/transpmodel.f90 [hunk 3]
```diff
@@ -1097,38 +1102,50 @@ subroutine vom_daily_init ()
       jmax25g_d(:) = jmax25g_d(2) * (/1.0d0-i_incrjmax,1.0d0,1.0d0+i_incrjmax/)
       jmax25g_d(:) = MAX(jmax25g_d(:), 50.0d-6)
 
+!     * adjust lai-values trees
+      lai_lt(:) = lai_lt(2) * (/1.0d0-i_incrlait,1.0d0,1.0d0+i_incrlait/)
+      lai_lt(:) = MAX( lai_lt(:), 0.1d0 ) !minimum value, else lai doesn't pick up anymore
```
Comment (schymans) [inline VOM_Fortran/VOM-code/transpmodel.f90:1110]: You might want to set the minimum to a smaller value, perhaps even supplied in the input file, along with i_incrlait.

File: VOM_Fortran/VOM-code/transpmodel.f90 [hunk 4]
```diff
@@ -1097,38 +1102,50 @@ subroutine vom_daily_init ()
...
+
+!     * adjust lai-values grasses
+      lai_lg(:) = lai_lg(2) * (/1.0d0-i_incrlaig,1.0d0,1.0d0+i_incrlaig/) 
+      lai_lg(:) = MAX( lai_lg(:), 0.1d0 ) !minimum value, else lai doesn't pick up anymore
```
Comment (schymans) [inline VOM_Fortran/VOM-code/transpmodel.f90:1114]: Same here, the minimum value could be supplied externally.

File: VOM_Fortran/VOM-code/transpmodel.f90 [hunk 5]
```diff
@@ -1185,10 +1204,17 @@ subroutine vom_hourly_init ()
      &           + i_hd) * jmax25t_d(:)) / ((-1.d0 + p_E ** ((i_hd     &
      &           * (273.d0 + tair_h(th_) - topt_)) / (tair_h(th_)      &
      &           + 273.d0 * p_R_ * topt_))) * i_ha + i_hd)
-!     * (3.24), (Out[312])
-      rlt_h(:) = ((ca_h(th_) - gammastar) * o_pct * jmaxt_h(:)         &
+
+!       * fraction of absorbed radiation per crown area (Beer-lambert)
+        Ma_lt(:) = 1.0d0 - p_E ** (-lai_lt(:)/2.0d0)
+
```
Comment (schymans) [inline VOM_Fortran/VOM-code/transpmodel.f90:1221]: Please reflect in the definition of Ma_lt that it is actually per crown area. Don't you need to include cases for the old behaviour and the new behaviour including LAI?

File: VOM_Fortran/VOM-code/transpmodel.f90 [hunk 6]
```diff
@@ -1169,7 +1186,9 @@ subroutine vom_hourly_init ()
       use vom_vegwat_mod
       implicit none
 
-      INTEGER :: ii
+      INTEGER :: ii           !counter
+      REAL*8 :: Ma_lg(3)      !fraction of absorbed radiation grasses
+      REAL*8 :: Ma_lt(3)      !fraction of absorbed radiation trees
```
Comment (schymans) [inline VOM_Fortran/VOM-code/transpmodel.f90:1197]: The way it is used below, this is the fraction of absorbed radiation per crown area.

File: VOM_Fortran/VOM-code/watbal.f90 [hunk 7]
```diff
@@ -189,7 +189,7 @@ subroutine waterbalance_fluxes ()
 
 !     * soil evaporation
 
-      esoil__  = 0.0002d0 * (1.d0 - 0.8d0 * (o_pct + pcg_d(2)))        &
+      esoil__  = 0.0002d0 * (1.d0 - 0.8d0 * (o_cai + pcg_d(2)))        &
```
Comment (schymans) [inline VOM_Fortran/VOM-code/watbal.f90:192]: Need to look into the original motivation for this formulation and then possibly replace o_cai+pcg_d by the fraction of absorbed radiation by both trees and grasses.

**Review by schymans — CHANGES_REQUESTED** [review]
Looks good, but need to document a bit more to make it more transparent what is happening.

File: VOM_Fortran/VOM-code/modules.f90 (hunk 1 above)
Comment (rcnijzink) [reply VOM_Fortran/VOM-code/modules.f90]: I guess this is something coming from the past... it was always like this for all parameters, but actually we can remove the numbers is these are overwritten anyway by either the input-files or the sce-parameters. I just changed here the name of o_pct into o_cai, that's all

File: VOM_Fortran/VOM-code/transpmodel.f90 (hunk 2 above)
Comment (rcnijzink) [reply VOM_Fortran/VOM-code/transpmodel.f90:990]: This is the subroutine that initializes everything, I setted the initial LAI-value to 2.5. Actually we should move this probably to the namelist as well, but for now I hard-coded it to see if it worked anyway.

File: VOM_Fortran/VOM-code/transpmodel.f90 (hunk 3 above)
Comment (rcnijzink) [reply VOM_Fortran/VOM-code/transpmodel.f90:1110]: Yes, good point, will do that.

File: VOM_Fortran/VOM-code/transpmodel.f90 (hunk 5 above)
Comment (rcnijzink) [reply VOM_Fortran/VOM-code/transpmodel.f90:1221]: There are already cases, but I guess you don's see that anymore as it is already in your master by https://github.com/schymans/VOM/pull/26

File: VOM_Fortran/VOM-code/transpmodel.f90 (hunk 6 above)
Comment (rcnijzink) [reply VOM_Fortran/VOM-code/transpmodel.f90:1197]: Yes, true, will document that.

File: VOM_Fortran/VOM-code/modules.f90 (hunk 1 above)
Comment (schymans) [reply VOM_Fortran/VOM-code/modules.f90]: Could you remove the numbers? Otherwise this is going to trip us and others up again.

Comment (rcnijzink) [reply VOM_Fortran/VOM-code/modules.f90]: Will do!

**Review by schymans — APPROVED** [review]
This is great, removal of hard-wired parameters from the code!

**Pull request merged**